### PR TITLE
[READY] - [inf-issue515] - term-apply config

### PR DESCRIPTION
**Pull request by kylerisse**
Adds a new server config type to term-apply.

Previous:

- environment variable parsing handled in both main and s3file packages
- hard coded s3 prefixes

New:

- environment variable parsing all handled in one place when creating new config (fixes inf issue 515)
- s3 prefixes can be changed with environment variables (prep-work to address inf issue 525)

Tests:
- [x] locally using live s3

### CODE REVIEW

File: apps/term-apply/pkg/applicant/manager.go [hunk 1]
```diff
@@ -14,26 +14,29 @@ type ApplicantManager struct {
...
 	writeChan  chan []applicant // serializes csv file writes
 	resumes    *resumeWatcher
+	bucket     string
+	csvKey     string
 }
 
-const DATA_FILE_KEY = "/term-apply/data/applicants.csv"
-
-func NewApplicantManager(filename, uploadDir string) (*ApplicantManager, error) {
+func NewApplicantManager(filename, uploadDir, bucket, resumePrefix, csvPrefix string) (*ApplicantManager, error) {
```
Comment (kylerisse) [inline apps/term-apply/pkg/applicant/manager.go:21]: not a fan of this function sig. smells like a refactor down the line.

File: apps/term-apply/pkg/server/config.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
+package server
+
+import (
+	"log"
+	"os"
+	"strconv"
+)
+
+type Config struct {
```
Comment (sarcasticadmin) [inline apps/term-apply/pkg/server/config.go:9]: Nice, Im really happy to see this land!

File: apps/term-apply/pkg/applicant/manager.go (hunk 1 above)
Comment (sarcasticadmin) [reply apps/term-apply/pkg/applicant/manager.go:21]: Ya make itd be worth the passing it the config? But I think its reasonable to leave it as-is

File: apps/term-apply/pkg/transfer/copy_from_client.go [hunk 3]
```diff
@@ -71,10 +75,9 @@ func (c *copyFromClientHandler) Write(s ssh.Session, entry *scp.FileEntry) (int6
 		return 0, fmt.Errorf("failed to write file: %q", entry.Filepath)
 	}
 
-	const RESUME_FILE_PREFIX = "/term-apply/resumes"
```
Comment (sarcasticadmin) [inline apps/term-apply/pkg/transfer/copy_from_client.go:74]: bye bye constant

**Review by sarcasticadmin — APPROVED** [review]
@kylerisse compile and ran locally (no submission) but everything seemed to be working great! Nice jcall refactoring this into a clean and concise config :shipit:

**Pull request merged**